src/btree/bt_read.c

```diff
@@ -372,9 +372,11 @@ __page_read(WT_SESSION_IMPL *session, WT_REF *ref)
 	 * the allocated copy of the disk image on return, the in-memory object
 	 * steals it.
 	 */
-	WT_ERR(__wt_page_inmem(session, ref, tmp.data,
-	    WT_DATA_IN_ITEM(&tmp) ?
-	    WT_PAGE_DISK_ALLOC : WT_PAGE_DISK_MAPPED, &page));
+	flags =
+	    WT_DATA_IN_ITEM(&tmp) ? WT_PAGE_DISK_ALLOC : WT_PAGE_DISK_MAPPED;
+	if (WT_SESSION_IS_CHECKPOINT(session))
+		LF_SET(WT_PAGE_CHECKPOINT_READ);
```

Comment:
This part doesn't work -- pages can be read as part of a checkpoint and stay in cache, so marking pages here isn't valid.  We could use `WT_SESSION_IS_CHECKPOINT` during the eviction if we want to exclude eviction of these pages for some reason.

That said, what was reported yesterday on the ticket was eviction of metadata triggering "progress" from the point of view of the eviction server, when nothing else could be evicted.  The problem with that is that we disable eviction checks when reading metadata, so metadata pages can be read back into cache when it is full.  That is why we need special treatment of metadata eviction (and any other eviction where reads disable ordinary eviction).